apps/src/studio/studio.js

```diff
@@ -3288,35 +3290,43 @@ Studio.animateGoals = function() {
   }
 };
 
+
 /**
- * Load clouds for the current background if it features them.
+ * Load clouds for the current background if it features them, or hide
+ * them if they shouldn't currently be shown.
  */
 Studio.loadClouds = function() {
+  var cloud, i;
   var showClouds = Studio.background && skin[Studio.background].clouds;
 
   var width, height;
   width = height = 300;
 
-  for (var i = 0; i < constants.NUM_CLOUDS; i++) {
-    // Start with clouds hidden.
-    var cloud = document.getElementById('cloud' + i);
-    cloud.setAttribute('x', -width);
-    cloud.setAttribute('y', -height);
-
-    // If we aren't showing clouds for this background, do nothing more.
-    if (!showClouds) {
-      continue;
+  if (!showClouds) {
+    // Hide the clouds offscreen.
+    for (i = 0; i < constants.NUM_CLOUDS; i++) {
+      cloud = document.getElementById('cloud' + i);
+      cloud.setAttribute('x', -width);
+      cloud.setAttribute('y', -height);
+    }
+  } else {
+    // Set up the right clouds.
+    for (i = 0; i < constants.NUM_CLOUDS; i++) {
```

Comment:
And here I would use `skin[Studio.background].clouds.length` so that you don't create more than are specified in the skin.